### ISSUE TITLE
[2.1.10] Swagger not working for multistore installs? 

### ISSUE THREAD

**Issue by zack6849** [magento/magento2#13474]
The default URL swagger attempts to fetch the details for the APIs is 

> http://base.url.example/rest/default/schema?services=all

The problem is, this doesn't work on our stores, Im not sure if magento's routing is expecting our default store's code to be `default`, or if it's a shortcut to the default store and not functioning for some other reason

For example, our store code for our default view is `mxs`, so our URL **should** be

> http://mikesxs.test/rest/mxs/schema?services=all

but swagger isn't picking up on the default store code.

### Preconditions

1. Magento v2.1.10
2. PHP v7.0.27
3. Ubuntu Linux 17.10 "Artful"

### Steps to reproduce

1.  Have a multistore setup (or maybe it just needs to have no store with the code `default`?)
2.  Navigate to ${base_url}/swagger/

### Expected result

1.  It should show endpoints.

### Actual result

1. It doesn't.

I wrote a small patch that i'm using in our local environment, that obviously isn't something i'd put in a PR because it uses the ObjectManager to get the current store code, but for anyone else that's having this issue, it can be fixed by doing the following

edit the magento-swagger module's file `view/frontend/templates/swagger-ui/index.phtml`

Change the following line:
`$schemaUrl = rtrim($block->getBaseUrl(), '/') . '/rest/default/schema?services=all';`

To this:

```
$objectManager = Magento\Framework\App\ObjectManager::getInstance();
$storeManager = $objectManager->get('\Magento\Store\Model\StoreManagerInterface');
$schemaUrl = rtrim($block->getBaseUrl(), '/') . '/rest/' . $storeManager->getStore()->getCode(). '/schema?services=all'; 
```

and refresh the page / flush any caches and you should be set.

**Comment by magento-engcom-team** [comment]
@zack6849, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by magento-engcom-team** [comment]
@zack6849 The issue is already fixed in `2.2` and `2.3-dev`.
Closing ticket for now. Please reopen if you need any further assistance with this.

**Comment by magento-engcom-team** [comment]
Hi @zack6849. Thank you for your report. 
The issue has been fixed in magento/magento2#13486 by @JeroenVanLeusden in 2.1-develop branch
Related commit(s):
 - [9e49ae79a3e023f38ce90e6fa0cab41edd62539d](https://github.com/magento/magento2/commit/9e49ae79a3e023f38ce90e6fa0cab41edd62539d)

The fix will be available with the upcoming 2.1.14 release.
